### DISCUSSION TITLE
How to connect front/back?

### DISCUSSION THREAD

**Discussion by ademDurakovic** [uNetworking/uWebSockets#1730] [Q&A] [answered]
So, I am trying to connect my front and back end right now but this Is my first time dealing with something like this and I don't really know where to begin.

Current I have a C++ file that looks like this and will act as my "backend"
```

#include <iostream>
#include "App.h"

int main() {
    uWS::App().get("/*", [](auto *res, auto *req) {
        // Enable CORS
        res->writeHeader("Access-Control-Allow-Origin", "*");
        res->writeHeader("Access-Control-Allow-Headers", "Origin, X-Requested-With, Content-Type, Accept");

        res->end("Hello World!");
    }).listen(9001, [](auto *listenSocket) {
        if (listenSocket) {
            std::cout << "Listening for connections..." << std::endl;
        }
    }).run();

    std::cout << "Shoot! We failed to listen and the App fell through, exiting now!" << std::endl;
}

```
Then I have a front end that does this on a button click:

`import React from "react";
import { useNavigate } from "react-router-dom";

function GameRoom() {
    const navigate = useNavigate();

    function minesButtonClick() {
        const socket = new WebSocket('ws://localhost:9001');

        socket.onopen = () => {
            console.log('Connected to the WebSocket backend');
            socket.send('Hello from client!');
        };

        socket.onmessage = (event) => {
            console.log('Received message from server:', event.data);
            // Handle the received message from the server
        };

        socket.onclose = () => {
            console.log('Disconnected from the WebSocket backend');
        };`

My front end console just logs me "bad response from server" disconnected message. 

Am I doing any of this correctly? I just want like an output in my backend saying like hey we have a connection.
Thanks.

**Comment by uNetworkingAB** [comment, marked as answer]
Use the EchoServer.cpp example for WebSockets instead of (what seems like) HelloWorld.cpp. EchoServer.cpp will work for your test

**Reply by ademDurakovic** [reply]
Your going to get a dedications/thank you slide on my final presentation.

Im able to print stuff in the front end now that I sent from the back. 

Thanks boss.
